src/erlcloud_iam.erl

```diff
@@ -664,17 +664,25 @@ get_instance_profile(ProfileName, #aws_config{} = Config) ->
 %
 % Account APIs
 %
--spec get_account_authorization_details() -> {ok, proplist()} |  {error, any()}.
+-spec get_account_authorization_details() -> {ok, proplist()} | {ok, proplist(), string()} | {error, any()}.
 get_account_authorization_details() ->
     get_account_authorization_details(default_config()).
   
--spec get_account_authorization_details(aws_config()) -> {ok, proplist()} |  {error, any()}.
+-spec get_account_authorization_details(list() | aws_config()) -> {ok, proplist()} | {ok, proplist(), string()} | {error, any()}.
 get_account_authorization_details(#aws_config{} = Config) ->
+    get_account_authorization_details([], #aws_config{} = Config);
+get_account_authorization_details(Params) ->
+    get_account_authorization_details(Params, default_config()).
+
+-spec get_account_authorization_details(list(), aws_config()) -> {ok, proplist()} | {ok, proplist(), string()} | {error, any()}.
+get_account_authorization_details(Params, #aws_config{} = Config) ->
```

Comment:
why not add a guard for a `when is_list(Params)` ?

Reply:
Done.